api/v1alpha1/backend_types.go

```diff
@@ -135,6 +142,16 @@ type BackendSpec struct {
 	Fallback *bool `json:"fallback,omitempty"`
 }
 
+// BackendType defines the type of the Backend.
+type BackendType string
+
+const (
+	// BackendTypeEndpoints defines the type of the backend as Endpoints.
+	BackendTypeEndpoints BackendType = "Endpoints"
+	// BackendTypeDynamicForwardProxy defines the type of the backend as DynamicForwardProxy.
+	BackendTypeDynamicForwardProxy BackendType = "DynamicForwardProxy"
```

Comment:
Nit : `DynamicForwardProxy` is the role of Envoy, should we call the Backend type `DynamicResolve` ?

Reply:
Yeah im open for any choice and agree DynamicForeqrdProxy is a bit confusing. How about DynamicResolver?

Reply:
DynamicResolver SGTM